ruby/ql/lib/codeql/ruby/ast/internal/TreeSitter.qll

```diff
@@ -48,6 +53,38 @@ module Ruby {
     final override string getAPrimaryQlClass() { result = "ReservedWord" }
   }
 
+  /** Gets the file path of the given `node`. */
+  overlay[local]
+  pragma[nomagic]
+  string getRawFile(@ruby_ast_node node) {
```

Comment:
Could we use `@file` instead of `string` in these predicates? AFAICT it would make the code simpler and also clearer because the types will be more precise.